qubes/ext/block.py

```diff
@@ -105,29 +152,208 @@ def size(self):
 
     @property
     def device_node(self):
-        '''Device node in backend domain'''
+        """Device node in backend domain"""
         return '/dev/' + self.ident.replace('_', '/')
 
+    @property
+    def interfaces(self) -> List[qubes.device_protocol.DeviceInterface]:
+        """
+        List of device interfaces.
+
+        Every device should have at least one interface.
+        """
+        return [qubes.device_protocol.DeviceInterface("******", "block")]
+
+    @property
+    def parent_device(self) -> Optional[qubes.device_protocol.Device]:
+        """
+        The parent device, if any.
+
+        If the device is part of another device (e.g., it's a single
+        partition of an usb stick), the parent device id should be here.
+        """
+        if self._parent is None:
+            if not self.backend_domain.is_running():
+                return None
+            untrusted_parent_info = self.backend_domain.untrusted_qdb.read(
+                f'/qubes-block-devices/{self.ident}/parent')
+            if untrusted_parent_info is None:
+                return None
+            # '4-4.1:1.0' -> parent_ident='4-4.1', interface_num='1.0'
+            # 'sda' -> parent_ident='sda', interface_num=''
+            parent_ident, sep, interface_num = self._sanitize(
+                untrusted_parent_info).partition(":")
+            devclass = 'usb' if sep == ':' else 'block'
```

Comment:
Looking at it now, since this is new code (so, no need to keep compatibility with some specific API), isn't it better for VM to explicitly say which device class parent is, instead of guessing based on ident format? I'm not sure what else it could be here, and USB vs block idents are quite different, but explicit info might be easier to handle.

Reply:
IIUC we prefer to have information like `usb:ident:interface` or `block:ident` in qubesdb? For now, I have proposed something in between, i.e., I use info from qubesdb to find device exposed by the VM

Reply:
IMO the (parent) device class should be explicit. It can be a prefix like you say here, or a separate qubesdb entry like it's in the DeviceInfo object serialization. The former is probably nicer?